sormas-stats-sitrep/utils/caseExport.R

```diff
@@ -0,0 +1,190 @@
+
+    FROM public.person
+
+    WHERE id IN (SELECT person_id FROM public.cases
+      WHERE deleted = FALSE 
+      and caseclassification != 'NO_CASE'
+      and reportdate between '", fromDate, "' and '", toDate, "' 
+      and disease = 'CORONAVIRUS')
+                         ")
+  # Only persons with a person_id that also appears in a case 
+  # are queried from the database
+  persons <- dbGetQuery(sormas_db,queryPersons)
+  
+  # Symptoms table SQL query
+  querySymptoms <- paste0("SELECT DISTINCT id AS symptoms_id,
+    onsetDate AS onset_date
+    
+    FROM public.symptoms
+    
+    WHERE id IN (SELECT symptoms_id FROM public.cases
+      WHERE deleted = FALSE 
+      and caseclassification != 'NO_CASE'
+      and reportdate between '", fromDate, "' and '", toDate, "' 
+      and disease = 'CORONAVIRUS')
+                          ")
+  # Only symptoms with a symptoms_id that also appears in a case 
+  # are queried from the database
+  symptoms <- dbGetQuery(sormas_db, querySymptoms)
+  
+  # Hospitalizations table query 
+  queryHospitalizations <- paste0("SELECT DISTINCT id AS hospitalization_id,
+    admittedToHealthFacility AS admitted_to_health_facility,
+    admissionDate AS admission_date,
+    hospitalizationReason AS hospitalization_reason
+    
+    FROM public.hospitalization
+    
+    WHERE id in (SELECT hospitalization_id FROM public.cases
+      WHERE deleted = FALSE 
+      and caseclassification != 'NO_CASE'
+      and reportdate between '", fromDate, "' and '", toDate, "' 
+      and disease = 'CORONAVIRUS')
+                            ")
+  # Only phospitalizations with a hospitalization_id that also appears in a case 
+  # are queried from the database
+  hospitalizations <- dbGetQuery(sormas_db, queryHospitalizations)
+  
+  # District table SQL query
+  queryDistrict <- paste0("SELECT DISTINCT id AS district_id,
+    name AS district_name
+    
+    FROM public.district
+    
+    WHERE archived = FALSE
+                          ")
+  districts <- dbGetQuery(sormas_db, queryDistrict)
+  
+  # Region table SQL query
+  queryRegions <- paste0("SELECT DISTINCT id AS region_id,
+    name AS region_name
+    
+    FROM public.region 
+    
+    WHERE archived = FALSE
+                         ")
+  regions <- dbGetQuery(sormas_db, queryRegions)
+  
+  # Populationdata table SQL query 
+  queryPopulation <- paste0("SELECT DISTINCT id AS population_id,
+                            district_id,
+                            region_id, 
+                            sex AS sex_population,
+                            agegroup AS agegroup_population,
+                            population
+                            
+                            FROM public.populationdata
+                            ")
+  population_data <- dbGetQuery(sormas_db, queryPopulation)
+  
+  # Merging population data with district names
+  population_data <- population_data %>% 
+    dplyr::left_join(., districts, by = 'district_id' )
```

Comment:
- I am not sure that this is possible since population data already contain the names of districts and  region, not their ids
- Better to remove anything about population from this function and we can discuss later how to go about it. Its tricky.
- The merging would be based on the region and district name and also account for duplicate if they exist.